### PR TITLE
Remove "SD cards" from strings Fix #12777

### LINKED ISSUE #12777: "SD card" used to mean "storage"
Let's not use `SD Card`, I don't believe this is accurate

_Originally posted by @david-allison in https://github.com/ankidroid/Anki-Android/pull/12683#discussion_r1002538445_
      

Actually, there are other strings that seems problematic too. 
```xml
    <string name="backup_deck_no_space_left">Backup not saved. Not enough space left on SD card. Lower the backup depth or remove some other files.</string>
    <string name="sd_space_warning">Less than %d MB space on your SD card left.\nFree some space to avoid data loss.</string>
    <string name="sync_write_access_error">The downloaded file couldn’t be saved to the SD card. Either the card is mounted read-only or there isn’t enough space.</string>
    <string name="sd_card_full_title">SD card full</string>
    <string name="sd_card_almost_full_title">SD card almost full</string>
```

This issue is resolved when those five strings are rephrased in a way that does not refer to SD card. Ideally, we should just remain vague and and mention space without even trying to indicate to the user which storage we are referring to exactly.
Since those will be new string, they require new translation, so we should also change the key of all of those strings.

For the record, original issue considered three options:
* we create a new string key, to get them retranslated
* we correct in English but know it won't be corrected in other locales
* We don't do anything because it's really no big deal

And option 1 was selected, as seen below

### PR DESCRIPTION
**Pull request by Subhadiptech**
As intended, I have followed the first preference as suggested by @mikehardy  and @Arthur-Milchior  and made the following changes.

Changes include creating new keys for the strings as follows:

    <string name="backup_deck_no_space_left">Backup not saved. Not enough space left on SD card. Lower the backup depth or remove some other files.</string>
    <string name="sd_space_warning">Less than %d MB space on your SD card left.\nFree some space to avoid data loss.</string>
    <string name="sync_write_access_error">The downloaded file couldn’t be saved to the SD card. Either the card is mounted read-only or there isn’t enough space.</string>
    <string name="sd_card_full_title">SD card full</string>
    <string name="sd_card_almost_full_title">SD card almost full</string>
    
    
The term "SD Card" is now Storage for the users on the applications.

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
Thanks @Subhadiptech 

A few requests. 
As you may see above, the "Code quality checks /lint debug" fails
```
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/res/values/03-dialogs.xml:101: Error: The resource R.string.storage_full_title appears to be unused [UnusedResources]
    <string name="storage_full_title">Storage full</string>
            ~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/res/values/03-dialogs.xml:104: Error: The resource R.string.storage_almost_full_title appears to be unused [UnusedResources]
    <string name="storage_almost_full_title">Storage almost full</string>
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/res/values/04-network.xml:69: Error: The resource R.string.sync_write_access_error_on_storage appears to be unused [UnusedResources]
    <string name="sync_write_access_error_on_storage">The downloaded file couldn’t be saved to your storage. Either the storage is mounted read-only or there isn’t enough space.</string>
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/res/values/09-backup.xml:24: Error: The resource R.string.backup_deck_no_storage_left appears to be unused [UnusedResources]
    <string name="backup_deck_no_storage_left">Backup not saved. Not enough space left on Storage. Lower the backup depth or remove some other files.</string>
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/res/values/09-backup.xml:27: Error: The resource R.string.storage_warning appears to be unused [UnusedResources]
    <string name="storage_warning">Less than %d MB space on your storage left.\nFree some space to avoid data loss.</string>
```

Could you please correct that. 
https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#lint explains how to run it on your device.
Essentially, it means that since a string is not used anymore, you should delete it from `values` directory.

(That also means you don't need to add comments to explain the difference)

As discussed on Discord, you should use those new strings in the code. The goal being for the user to see them.

One last remark. I'd appreciate if the commit message and the PR message could explains in plain English what is being done.
The goal is that devs can look at the history of the file, see when a line was introduced, and what was the reason. Devs won't know what bug number 12777 is. However, if they read 
> Remove "SD cards" from strings

they'll probably get a guess, at least understand that it's just changing a string value; so if they are looking for a bug, know that it was probably not related to this change.

**Comment by Subhadiptech** [comment]
Ya, I have used the strings now. Just changing JDK version and pushing

**Review by Arthur-Milchior — APPROVED** [review]
Thanks

**Comment by Arthur-Milchior** [comment]
It would be nice if you could put all of this in a single commit, and put a descriptive commit message, as I explained above. Don't hesitate to force push on your own branch

**Comment by Subhadiptech** [comment]
@Arthur-Milchior I will create new branch from next time. Just wanted to ask how to merge all the commits into a single commit?

**Comment by Arthur-Milchior** [comment]
You answered "sure", so I guess I'll wait for this last change to be done.

File: AnkiDroid/src/main/res/values/09-backup.xml [hunk 1]
```diff
@@ -18,9 +18,9 @@
 ~ You should have received a copy of the GNU General Public License along with
 ~ this program.  If not, see <http://www.gnu.org/licenses/>.
 --><resources>
-
-    <string name="backup_deck_no_space_left">Backup not saved. Not enough space left on SD card. Lower the backup depth or remove some other files.</string>
-    <string name="sd_space_warning">Less than %d MB space on your SD card left.\nFree some space to avoid data loss.</string>
+    
+    <string name="backup_deck_no_storage_left">Backup not saved. Not enough space left on Storage. Lower the backup depth or remove some other files.</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/09-backup.xml:22]:
```suggestion
    <string name="backup_deck_no_storage_left">Backup not saved. Not enough space left on your storage. Lower the backup depth or remove some other files.</string>
```

File: AnkiDroid/src/main/res/values/04-network.xml [hunk 2]
```diff
@@ -64,7 +64,7 @@
     <string name="sync_remote_db_error">The downloaded database was corrupt. Please try again.</string>
     <string name="sync_overwrite_error">Your current collection couldn’t be overwritten by the downloaded file</string>
     <string name="sync_connection_error">Connection timed out. Either your internet connection is experiencing problems, or you have a very large file in your media folder.</string>
-    <string name="sync_write_access_error">The downloaded file couldn’t be saved to the SD card. Either the card is mounted read-only or there isn’t enough space.</string>
+    <string name="sync_write_access_error_on_storage">The downloaded file couldn’t be saved to your storage. Either the storage is mounted read-only or there isn’t enough space.</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/04-network.xml:67]:
```suggestion
    <string name="sync_write_access_error_on_storage">The downloaded file couldn’t be saved to your storage. Either the storage is read-only or there isn’t enough space.</string>
```

**Review by david-allison — CHANGES_REQUESTED** [review]
Thanks! Couple of nitpicks

**Comment by david-allison** [comment]
Pending squash merge and sync from https://github.com/ankidroid/Anki-Android/actions/runs/3420021758

**Review by mikehardy — APPROVED** [review]
Looks good to me - and I've just cleared strings out via crowdin sync + merged translations PR
